src/EditorFeatures/CSharp/InlineRename/CSharpEditorInlineRenameService.cs

```diff
@@ -18,4 +26,105 @@ internal sealed class CSharpEditorInlineRenameService(
     [ImportMany] IEnumerable<IRefactorNotifyService> refactorNotifyServices,
     IGlobalOptionService globalOptions) : AbstractEditorInlineRenameService(refactorNotifyServices, globalOptions)
 {
+    protected override async Task<ImmutableDictionary<string, ImmutableArray<string>>> GetRenameContextCoreAsync(IInlineRenameInfo inlineRenameInfo, IInlineRenameLocationSet inlineRenameLocationSet, CancellationToken cancellationToken)
+    {
+        var seen = PooledHashSet<TextSpan>.GetInstance();
+        var definitions = ArrayBuilder<string>.GetInstance();
+        var references = ArrayBuilder<string>.GetInstance();
+        var docComments = ArrayBuilder<string>.GetInstance();
+
+        foreach (var renameDefinition in inlineRenameInfo.DefinitionLocations)
+        {
+            var containingStatementOrDeclarationSpan =
+                await renameDefinition.Document.TryGetSurroundingNodeSpanAsync<StatementSyntax>(renameDefinition.SourceSpan, cancellationToken).ConfigureAwait(false) ??
+                await renameDefinition.Document.TryGetSurroundingNodeSpanAsync<MemberDeclarationSyntax>(renameDefinition.SourceSpan, cancellationToken).ConfigureAwait(false);
```

Comment:
note: this may be huge.  is something limiting how much we actually consume?

Reply:
didn't see a response to this.

Reply:
It looks like the max number is limited to 10.